RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,1030 @@
+    unsigned int k=critical[ic].second;
+
+    // estimate subcluster positions and weight
+    double p1=0, z1=0, t1=0, w1=0;
+    double p2=0, z2=0, t2=0, w2=0;
+    for(unsigned int i=0; i<nt; ++i){
+      if (tks.Z_sum_[i] > 1.e-100) {
+	// winner-takes-all, usually overestimates splitting
+	double tl = tks.z_[i] < y.z_[k] ? 1.: 0.;
+	double tr = 1. - tl;
+
+	// soften it, especially at low T
+	/*
+	double arg = ( tks.z_[i] - y.z_[k] ) * sqrt(beta * tks.dz2_[i]); // + std::fabs(tks.t_[i] - y.t_[k])
+	std::cout << arg << std::endl;
+	if(std::fabs(arg) < 20){
+	  double t = local_exp(-arg);
+	  tl = t/(t+1.);
+	  tr = 1/(t+1.);
+	}
+	*/
+
+	double p = y.pk_[k] * tks.pi_[i] * local_exp(-beta * Eik(tks.z_[i], y.z_[k], tks.dz2_[i], 
+                                                                 tks.t_[i], y.t_[k], tks.dt2_[i])) / tks.Z_sum_[i];
+	double w = p*tks.errsum_[i];
+	p1 += p*tl;  z1 += w*tl*tks.z_[i]; t1 += w*tl*tks.t_[i]; w1 += w*tl;
+	p2 += p*tr;  z2 += w*tr*tks.z_[i]; t2 += w*tr*tks.t_[i]; w2 += w*tr;
+      }
+    }
+
+    if(w1>0){z1 = z1/w1; t1 = t1/w1;} else {z1=y.z_[k]-epsilonz; t1=y.t_[k]-epsilont;}
+    if(w2>0){z2 = z2/w2; t2 = t2/w2;} else {z2=y.z_[k]+epsilonz; t2=y.t_[k]+epsilont;}
+    
+    // reduce split size if there is not enough room
+    if( ( k   > 0 ) && ( z1 < (0.6*y.z_[k] + 0.4*y.z_[k-1])) ){ 
+      z1 = 0.5*y.z_[k] + 0.5*y.z_[k-1]; 
+      t1 = 0.5*y.t_[k] + 0.5*y.t_[k-1]; 
+    }
+    if( ( k+1 < nv) && ( z2 > (0.6*y.z_[k] + 0.4*y.z_[k+1])) ){ 
+      z2 = 0.5*y.z_[k] + 0.5*y.z_[k+1]; 
+      t2 = 0.5*y.t_[k] + 0.5*y.t_[k+1]; 
+    }
+    
+#ifdef VI_DEBUG
+    if(verbose_){
+      if (std::fabs(y.z_[k] - zdumpcenter_) < zdumpwidth_){
+	std::cout << " T= " << std::setw(8) << 1./beta 
+		  << " Tc= " << critical[ic].first 
+		  << "    splitting " << std::fixed << std::setprecision(4) << y.z_[k] 
+		  << " --> (" << z1 << ',' << t1<< "),(" << z2 << ',' << t2 
+		  << ")     [" << p1 << "," << p2 << "]" ;
+	if (std::fabs(z2-z1) > epsilonz || std::fabs(t2-t1) > epsilont){
+	  std::cout << std::endl;
+	}else{
+	  std::cout <<  "  rejected " << std::endl;
+	}
+      }
+    }
+#endif
+
+    // split if the new subclusters are significantly separated
+    if( std::fabs(z2-z1) > epsilonz || std::fabs(t2-t1) > epsilont){
+      split = true;
+      double pk1 = p1*y.pk_[k]/(p1+p2);
+      double pk2 = p2*y.pk_[k]/(p1+p2);
+      y.z_[k]  =  z2;
+      y.t_[k]  =  t2;
+      y.pk_[k] = pk2;
+      y.insertItem(k, z1, t1, pk1);
+      nv++;
+
+     // adjust remaining pointers
+      for(unsigned int jc=ic; jc < critical.size(); jc++){
+        if (critical[jc].second > k) {critical[jc].second++;}
+      }
+    }
+  }
+  return split;
+}
+
+
+
+void DAClusterizerInZT_vect::splitAll( vertex_t & y) const {
+
+  const unsigned int nv = y.getSize();
+  
+  constexpr double epsilonz = 1e-3; // split all single vertices by 10 um
+  constexpr double epsilont = 1e-2; // split all single vertices by 10 ps
```

Comment:
it looks like these are repeated as in ```split``` method.
Is this ad hoc and can be different or should this be made to a common constant?